mtls-northbound/docs/cloudshell-tutorial.md

```diff
@@ -0,0 +1,586 @@
+```
+Sample response:
+```
+{
+    "request":"GET https://xapi-dev.kurtkanaskie.net/v1/samples/mtls",
+    "status":"200",
+    "reason":"OK",
+    "organization":"apigeex-mint-kurt",
+    "environment":"dev",
+    "tls_protocol":"TLSv1.3",
+    "tls_cipher":"TLS_AES_128_GCM_SHA256",
+    "tls_server.name":"xapi-dev.kurtkanaskie.net",
+    "tls_session.id":"",
+    "mtls_details":{
+        "x-client-cert-error":"",
+        "x-client-cert-present":"true",
+        "x-client-cert-chain-verified":"true",
+        "x-client-cert-serial-number":"00:ad:3e:0e:bc:c3:d4:f4:2d:ae:5b:21:3b:e4:cb:8b:f4:9d:49:a1",
+        "x-client-cert-hash":"om8vGt3r6eTx5TI1kCeNKRJYZTUreIbHGu+Gu8FN8To",
+        "x-client-cert-dnsname-sans":"eGFwaS1kZXYua3VydGthbmFza2llLm5ldA==",
+        "x-client-cert-dnsname-sans-decoded":"xapi-dev.kurtkanaskie.net",
+        "x-client-cert-uri-sans":"aHR0cHM6Ly94YXBpLWRldi5rdXJ0a2FuYXNraWUubmV0",
+        "x-client-cert-uri-sans-decoded":"https://xapi-dev.kurtkanaskie.net",
+        "x-client-cert-spiffe":"",
+        "x-client-cert-valid-not-before":"2023-09-12T14:55:19+00:00",
+        "x-client-cert-valid-not-after":"2023-10-12T14:55:18+00:00"
+    }
+}
+```
+
+#### Test with invalid certificate and key
+Now lets test with the invalid certificate and key.\
+Again, notice the response is from curl, not the Raise Fault policy in the proxy.
+
+```
+curl https://$APIGEE_HOST/v1/samples/mtls --cert ./${INVALID_CERT_NAME}-cert.pem --key ./${INVALID_CERT_NAME}-key.pem
+```
+Sample response:
+```
+curl: (56) Recv failure: Connection reset by peer
+```
+
+### Check GLB Logs
+Since we have enforced strict mTLS and our requests are being rejected by the GLB, lets take a look at Cloud Logging to see the errors there.
+
+GLB Failures details [here](https://cloud.google.com/load-balancing/docs/https/https-logging-monitoring#failure-messages). \
+How top view logs [documentation](https://cloud.google.com/load-balancing/docs/https/https-logging-monitoring#log-fwd-rule).
+
+**NOTE:** Log entries will only exist when clientValidationMode is set to REJECT_INVALID and not ALLOW_INVALID_OR_MISSING_CLIENT_CERT.
+
+Find the forwarding rule for the GLB. For example:
+```
+gcloud compute forwarding-rules list
+```
+Sample response:
+```
+NAME                                              REGION    IP_ADDRESS      IP_PROTOCOL  TARGET
+apigee-proxy-https-lb-rule                  34.149.167.159  TCP          apigee-proxy-https-proxy
+apigee-proxy-modern-https-lb-rule           34.160.201.100  TCP          apigee-proxy-modern-https-proxy
+```
+
+Open a separate tab and navigate to Cloud Logging in the GCP Console - (https://console.cloud.google.com/logs/query). 
+
+Enter the query for you forwarding rule.
+```
+jsonPayload.statusDetails=~"client_cert"
+jsonPayload.@type="type.googleapis.com/google.cloud.loadbalancing.type.LoadBalancerLogEntry"
+resource.labels.forwarding_rule_name=apigee-proxy-https-lb-rule
```

Comment:
Make it clear they need to replace this with the name of the forwarding rule from their project